portfolio/src/main/webapp/mapScript.js

```diff
@@ -0,0 +1,154 @@
+    center: unc,
+    zoom: 14
+  });
+
+    map.addListener('click', (event) => {
+        createMarkerForEdit(event.latLng.lat(), event.latLng.lng());
+    });
+
+    fetchMarkers();
+}
+
+/** Fetches markers from the backend and adds them to the map. */
+const fetchMarkers = function() {
+    fetch('/markers').then(response => response.json()).then((markers) => {
+        markers.forEach((marker) => {
+            createMarkerForDisplay(marker.lat, marker.lng, marker.title, marker.description);
+            $('#place-list').append('<li>' + marker.title + '</li>');
+        });
+    });
+}
+
+/** Creates a marker that shows a read-only info window when clicked. */
+const createMarkerForDisplay = function(lat, lng, title, description) {
+    const marker = new google.maps.Marker({position: {lat: lat, lng: lng}, map: map, title: title});
+    const content = '<div><h1>'+ title +'</h1></div>' + 
+                    '<div><p>' + description + '</p></div>';
+    const infoWindow = new google.maps.InfoWindow({content: content});
+    marker.addListener('click', () => {
+        infoWindow.open(map, marker);
+    });
+}
+
+/** Sends a marker to the backend for saving. */
+const postMarker = function(lat, lng, title, description) {
+    const params = new URLSearchParams();
+    params.append('lat', lat);
+    params.append('lng', lng);
+    params.append('title', title);
+    params.append('description', description);
+
+    fetch('/markers', {method: 'POST', body: params});
+}
+
+/** Creates a marker that shows a textbox the user can edit. */
+createMarkerForEdit = function(lat, lng) {
+    // If we're already showing an editable marker, then remove it.
+    if (editMarker) {
+        editMarker.setMap(null);
+    }
+
+    editMarker = new google.maps.Marker({position: {lat: lat, lng: lng}, map: map});
+    const infoWindow = new google.maps.InfoWindow({content: buildInfoWindowInput(lat, lng)});
+
+    // When the user closes the editable info window, remove the marker.
+    google.maps.event.addListener(infoWindow, 'closeclick', () => {
+        editMarker.setMap(null);
+    });
+    infoWindow.open(map, editMarker);
+}
+
+/**
+ * Builds and returns HTML elements that show an editable textbox and a submit
+ * button.
+ */
+const buildInfoWindowInput = function(lat, lng) {
+    const title = document.createElement('input');
+    const description = document.createElement('textarea');
+    const button = document.createElement('button');
+    button.appendChild(document.createTextNode('Submit'));
+
+    button.onclick = () => {
+        postMarker(lat, lng, title.value, description.value);
+        createMarkerForDisplay(lat, lng, title.value, description.value);
+        editMarker.setMap(null);
+    };
+
+    const containerDiv = document.createElement('div');
+    containerDiv.appendChild(title);
+    containerDiv.appendChild(description);
+    containerDiv.appendChild(document.createElement('br'));
```

Comment:
Avoid using br for styling.

On the whole, you may want to style some of this - for example, the title and description inputs aren't labelled, so it's not very clear to a user what they're supposed to do here.

Reply:
I don't think the br is doing anything differently for me? Testing locally, it looks the same to me with and without the br. I think you can just take it out?